raster/r.in.pdal/point_binning.c

```diff
@@ -57,7 +57,7 @@ int blank_array(void *array, int nrows, int ncols, RASTER_MAP_TYPE map_type,
     case -1:
         /* fill with NULL */
         /* alloc for col+1, do we come up (nrows) short? no. */
-        Rast_set_null_value(array, nrows * ncols, map_type);
+        Rast_set_null_value(array, (size_t)nrows * ncols, map_type);
```

Comment:
`size_t` on 64 bit systems is larger than `unsigned int`, this will not work for larger regions.

I would rather loop over the rows and set each row to NULL. A pointer can be used to go over the rows, increased with `ptr = G_incr_void_ptr(ptr, ncols * map_type_size)`:

```
        for (row = 0; row < nrows; row++) {
            Rast_set_null_value(ptr, ncols, map_type);
            ptr = G_incr_void_ptr(ptr, ncols * Rast_cell_size(map_type));
        }
```

Reply:
While "set whole raster map to null" seems like a good function to have, I'm afraid that the correct way of using _Rast_set_null_value_ is to set only up a row of value as @metzm  says.

Reply:
With the change @metzm suggested, I guess there is no need to change (at this time) `Rast_set_null_value()`?

Reply:
"Set whole raster map to null" would be a rather special function because typically raster data are accessed by rows and not as a whole. Standard procedure would be e.g. to set the next output row to NULL, do some calculations with input data, set data in the output row where applicable, write output row, continue with next row.

In order to save memory consumption, raster operations should wherever possible never load a whole raster to memory but instead proceed row by row or a small chunk of rows (e.g. r.neighbors). This is IMHO a big strength of GRASS, also for cloud and grid computing.

Reply:
Just pushed the suggested changes. Please review!